src/hooks/accounting.ts

```diff
@@ -1,57 +1,87 @@
+import { LastCashoutActionResponse, PeerBalance, Settlements } from '@ethersphere/bee-js'
 import { useEffect, useState } from 'react'
 import { beeDebugApi } from '../services/bee'
 import { useApiPeerBalances, useApiSettlements } from './apiHooks'
 
 interface UseAccountingHook {
   isLoading: boolean
   isLoadingUncashed: boolean
+  error: Error | null
   totalsent: number
   totalreceived: number
-  accounting: Record<string, Accounting> | null
+  accounting: Accounting[] | null
 }
 
-interface CashoutInfo {
-  uncashedAmount: number
-  peer: string
-}
+/**
+ * Merges the balances, settlements and uncashedAmounts arrays into single array which is sorted by uncashed amounts (if any)
+ *
+ * @param balances         Balances for all peers
+ * @param settlements      Settlements for all peers which has some settlement
+ * @param uncashedAmounts  Array of getPeerLastCashout responses which is needed to calculate uncashed amount
+ *
+ * @returns
+ */
+function mergeAccounting(
+  balances?: PeerBalance[],
+  settlements?: Settlements[],
+  uncashedAmounts?: LastCashoutActionResponse[],
+): Accounting[] | null {
+  // Settlements or balances are still loading or there is an error -> return null
+  if (!balances || !settlements) return null
 
-export const useAccounting = (): UseAccountingHook => {
-  const settlements = useApiSettlements()
-  const balances = useApiPeerBalances()
   const accounting: Record<string, Accounting> = {}
-  const [isLoadingUncashed, setIsloadingUncashed] = useState<boolean>(false)
-  const [uncashedAmounts, setUncashedAmounts] = useState<CashoutInfo[] | null>(null)
 
-  balances.peerBalances?.balances.forEach(
-    // Some peers may not have settlement but all have balance (therefore initialize sent + received to 0)
-    ({ peer, balance }) => (accounting[peer] = { balance, sent: 0, received: 0, uncashedAmount: 0 }),
+  balances.forEach(
```

Comment:
```suggestion
  balances?.forEach(
```

Reply:
No need, the balances can not be null here.